### PR TITLE
Add has_voting_exclusions flag to cluster health output

### PR DESCRIPTION
**Pull request by andrershov**
This PR adds "has_voting_exclusions" flag to cluster health output.
Today voting exclusions can be already obtained by inspecting cluster_state.metadata.coordination_metadata. However, we want to make them visible, because incorrectly configured voting exclusions can render cluster non-functional. Having voting exclusions temporary is ok, however, once desired nodes are stopped, we expect users to remove them from the cluster state.
Cluster health output is the first thing users are looking at and sharing with others when trying to troubleshoot, so we decided to include this information in cluster health.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/health/ClusterStateHealth.java [hunk 1]
```diff
@@ -134,6 +137,11 @@ public ClusterStateHealth(final StreamInput in) throws IOException {
         unassignedShards = in.readVInt();
         numberOfNodes = in.readVInt();
         numberOfDataNodes = in.readVInt();
+        if (in.getVersion().onOrAfter(Version.V_7_0_0)) {
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/cluster/health/ClusterStateHealth.java]: By the time this lands, `master` will be `V_8_0_0`, and you will have to do the BWC dance to backport it to `7.x` and `7.0`. I also think this counts as a feature and therefore it shouldn't be backported to `7.0`?

The same comment applies to other places that look like this.

File: docs/java-api/admin/cluster/health.asciidoc [hunk 2]
```diff
@@ -10,26 +10,28 @@ some technical information about the cluster status per index:
...
 <1> Get information for all indices
 <2> Access the cluster name
 <3> Get the total number of data nodes
 <4> Get the total number of nodes
-<5> Iterate over all indices
-<6> Index name
-<7> Number of shards
-<8> Number of replicas
-<9> Index status
+<5> Check if there are voting exclusions in the cluster state
```
Comment (DaveCTurner) [inline docs/java-api/admin/cluster/health.asciidoc]: Could `voting exclusions` here be a link to the section entitled `Voting configuration exclusions API`, and could you add a

```
NOTE: Clusters should have no voting configuration exclusions in normal operation.
```

to those API docs?

File: docs/java-rest/high-level/cluster/health.asciidoc [hunk 3]
```diff
@@ -137,6 +137,7 @@ include-tagged::{doc-tests-file}[{api}-response-nodes]
 --------------------------------------------------
 <1> Number of nodes in the cluster
 <2> Number of data nodes in the cluster
+<3> Whether cluster has voting exclusions
```
Comment (DaveCTurner) [inline docs/java-rest/high-level/cluster/health.asciidoc]: Could `voting exclusions` here be a link to the section entitled `Voting configuration exclusions API`?

**Review by DaveCTurner — COMMENTED** [review]
Looks good, I asked for a few docs changes and noted the BWC issue that we face here.

I also wonder, is it easy to only include this field in the JSON output if `true`, so that in the common case we wouldn't see any change?

File: server/src/main/java/org/elasticsearch/cluster/health/ClusterStateHealth.java (hunk 1 above)
Comment (andrershov) [reply server/src/main/java/org/elasticsearch/cluster/health/ClusterStateHealth.java]: Ok, I have done the following things:

1) Repurposed the PR to 8.0 and 7.1
2) Disabled bwc tests for PR that goes to 8.0
3) Added TODO to change V_7_0_0 once https://github.com/elastic/elasticsearch/pull/38513 is merged

File: docs/java-api/admin/cluster/health.asciidoc (hunk 2 above)
Comment (andrershov) [reply docs/java-api/admin/cluster/health.asciidoc]: https://github.com/elastic/elasticsearch/pull/38568/commits/0c28c0f5cf312a835843918a11a03b42102c1dce

File: docs/java-rest/high-level/cluster/health.asciidoc (hunk 3 above)
Comment (andrershov) [reply docs/java-rest/high-level/cluster/health.asciidoc]: https://github.com/elastic/elasticsearch/pull/38568/commits/0c28c0f5cf312a835843918a11a03b42102c1dce

File: server/src/main/java/org/elasticsearch/cluster/health/ClusterStateHealth.java (hunk 1 above)
Comment (andrershov) [reply server/src/main/java/org/elasticsearch/cluster/health/ClusterStateHealth.java]: Now because #38513 is merged, I've changed the version to V_7_1_0

**Comment by andrershov** [comment]
@DaveCTurner it's ready for the second pass.

**Review by DaveCTurner — CHANGES_REQUESTED** [review]
On reflection I think we shouldn't do this, sorry. Although it might sometimes be very useful to see this flag in cluster health, I think we could probably say the same about many other settings. I think it's enough to expose this in the support diagnostics tooling as planned elsewhere.

**Pull request closed without merging**